vm-setup/roles/v1aX_integration_test/templates/cluster-template-controlplane-kubeadm-config-centos.yaml

```
@@ -15,6 +15,7 @@ preKubeadmCommands:
   - systemctl start monitor.keepalived.service
 postKubeadmCommands:
   - mkdir -p /home/{{ IMAGE_USERNAME }}/.kube
+  - chown {{ IMAGE_USERNAME }}:{{ IMAGE_USERNAME }} /home/{{ IMAGE_USERNAME }}/.kube
```

should we be specific and give permission to cache folder only?
```suggestion
  - chown {{ IMAGE_USERNAME }}:{{ IMAGE_USERNAME }} /home/{{ IMAGE_USERNAME }}/.kube/cache
```

---

As it is in the users home the whole directory should belong to the user. I do not see any reason why it should not be.
Generally from a users perspective I would not be amused if I cannot remove a directory in my home directory because it belongs to root.

---

Fair enough, also I just noticed that we already chown `.kube/config` in the same file (Line 20)

---

Yeah, that is still needed as the copy in line 19 runs as root as well and the config file gets created with root:root. I would leave it like this even though a
```
chown -R chown {{ IMAGE_USERNAME }}:{{ IMAGE_USERNAME }} /home/{{ IMAGE_USERNAME }}/.kube
```
replacing both current chown command would be possible, I think it is better to leave it as explicit as it is right now with the additional advantage of not having to use -R.